src/js/controller/captions.js

```diff
@@ -53,14 +56,45 @@ define([], function() {
 
         function _itemReadyHandler(item) {
             // Clean up in case we're replaying
-            _itemHandler(_model,item);
+            _itemHandler(_model, item);
 
-            // listen for tracks coming from the provider
-            _model.mediaController.on('subtitlesTracks', _subtitlesTracksHandler, this);
+            _model.mediaController.off('subtitlesTracks', _subtitlesTracksHandler, this);
 
-            var captionsMenu = _captionsMenu();
-            this.setCaptionsList(captionsMenu);
```

Comment:
This should not have been removed, or moved into a branch of code where it is only set sometimes. The model's captionsList should always be an array, and it must be reset for each item.

test/squash/bootstrap.html?primary=html5&edition=ads&feature=captions_basic

`jwplayer().getCaptionsList()` returns `undefined`, because the captions list is not set for the first item when the item changes.